markdown/src/org/intellij/plugins/markdown/structureView/MarkdownStructureViewFactory.java

```diff
@@ -29,19 +36,21 @@ public MarkdownStructureViewModel(@NotNull PsiFile psiFile, @Nullable Editor edi
       super(psiFile, editor, new MarkdownStructureElement(psiFile));
     }
 
+    @Nullable
     @Override
-    protected boolean isSuitable(PsiElement element) {
-      if (element == null) {
-        return false;
-      }
-      if (!MarkdownStructureElement.PRESENTABLE_TYPES.contains(element.getNode().getElementType())) {
-        return false;
-      }
-      final PsiElement parent = element.getParent();
-      if (MarkdownStructureElement.hasTrivialChild(parent)) {
-        return false;
+    protected Object findAcceptableElement(PsiElement element) {
+      // walk up the psi-tree until we find an element from the structure view
+      while (element != null && !PRESENTABLE_TYPES.contains(element.getNode().getElementType())) {
```

Comment:
Since the psi-tree generated by the current markdown parser is more a flat list than a tree, I think `getParentOfType` won't find the section element. But I may be wrong.